### PR TITLE
[LIBCLC][HIP] Add HIP AMD support for ilogb, log2, remainder

### PR DESCRIPTION
**Pull request by hdelan**
Fixing bug due to incorrect naming of remainder.cl as remainer.cl.

### CODE REVIEW

File: libclc/amdgcn-amdhsa/libspirv/math/log2.cl [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+#include <spirv/spirv.h>
+ 
+double __ocml_log2_f64(double);
+float __ocml_log2_f32(float);
+ 
+#define __CLC_FUNCTION __spirv_ocl_log2
+#define __CLC_BUILTIN __ocml_log2
+#define __CLC_BUILTIN_F __CLC_XCONCAT(__CLC_BUILTIN, _f32)
+#define __CLC_BUILTIN_D __CLC_XCONCAT(__CLC_BUILTIN, _f64)
+#include <math/unary_builtin.inc>
```
Comment (bader) [inline libclc/amdgcn-amdhsa/libspirv/math/log2.cl:19]: How about defining `__CLC_BUILTIN_H` for `log2` and `remainder` to make them consistent with `ilogb`, which is defined for `half` data type?

**Review by bader — DISMISSED** [review]

File: libclc/amdgcn-amdhsa/libspirv/math/log2.cl (hunk 1 above)
Comment (hdelan) [reply libclc/amdgcn-amdhsa/libspirv/math/log2.cl:20]: Change made. Just defaulting to `__CLC_BUILTIN_F`

Comment (bader) [reply libclc/amdgcn-amdhsa/libspirv/math/log2.cl:20]: Sorry, I missed that <math/unary_builtin.inc> defines function overload for half data type and uses float built-in by default to implement it.

**Review by bader — APPROVED** [review]

File: libclc/amdgcn-amdhsa/libspirv/math/log2.cl (hunk 1 above)
Comment (hdelan) [reply libclc/amdgcn-amdhsa/libspirv/math/log2.cl:20]: Sorry I missed that too. Will revert last commit

**Comment by AerialMantis** [comment]
@bader could we get another review on this one? I also see that one of the CI jobs failed but the failure doesn't seem related, would you be able to re-run the CI.

**Review by bader — APPROVED** [review]

**Comment by bader** [comment]
> @bader could we get another review on this one? I also see that one of the CI jobs failed but the failure doesn't seem related, would you be able to re-run the CI.

I did this a few weeks ago and it failed after re-run. I just re-started CI jobs again, let's see if it passes on third attempt. If not, I would recommend pulling `sycl` branch to https://github.com/hdelan/llvm/tree/hugh/libclc_amd.

I missed the review because it's doesn't show up on my GitHub dashboard - https://github.com/pulls/review-requested. I would appreciate if you can request via GitHub UI.
image
Let me know if it's not available for you. I usually track PRs where my review is explicitly requested. It's hard to track all PRs.

**Comment by bader** [comment]
The test still fails. @hdelan, could you update the branch, please?

**Comment by hdelan** [comment]
Branch updated! Sorry for delay

**Review by bader — APPROVED** [review]

**Comment by bader** [comment]
> I don't see that option, maybe only the author of the PR can see that?

Do you see that option in closed PR or you don't see in in open PRs as well? Do you see it here: https://github.com/intel/llvm/pull/5303?

**Comment by AerialMantis** [comment]
Okay yeah I can see it on that one, I did check another open PR but I think it only shows if there's been changes since that person's reviews, thanks :)

**Pull request merged**